// doggo-api-wrapper/index.js
const got = require('got')
const URI = 'https://dog.ceo/api'

class DogApi {
  constructor (baseURL = 'https://dog.ceo/api') {
    this.baseURL = baseURL
  }

  getListOfAllBreeds () {
    return got(`${this.baseURL}/breeds/list/all`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }

  getARandomDog () {
    return got(`${this.baseURL}/breeds/image/random`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }

  getMultipleRandomDogs (amount) {
    return got(`${this.baseURL}/breeds/image/random/${amount}`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }

  getAllDogsByBreed (breed) {
    return got(`${this.baseURL}/breed/${breed}/images`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }

  getARandomDogFromBreed (breed) {
    return got(`${this.baseURL}/breed/${breed}/images/random`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }

  getSubBreedFromBreed (breed) {
    return got(`${this.baseURL}/breed/${breed}/list`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }

  getAllSubBreedImages (breed, subBreed) {
    return got(`${this.baseURL}/breed/${breed}/${subBreed}/images`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }

  getARandomSubBreedImage (breed, subBreed) {
    return got(`${URI}/breed/${breed}/${subBreed}/images/random`, {
      json: true
    }).then(({ body }) => body).catch(err => err)
  }
}

module.exports = DogApi
